gaidds/Django-crm-project
# Create closed deals per month

### Closed Deals Count Per Month
**Summary:**

A new feature to endpoint `api/dashboard/` has been added.  It returns the count of closed deals per month, broken down into:

- Closed Won deals per month

- Closed Lost deals per month

- Total closed deals per month

To achieve this, a new column `real_close_date` was added to the Deal model. This column stores the actual date when the deal's stage changed to either "CLOSED WON" or "CLOSED LOST" (as opposed to the expected close date). Using `real_close_date`, the closed deals are grouped by month to generate the count.

**How to Test:**

1. Apply migrations:

Option 1: Run the following command in your terminal to restart the containers and apply migrations:
`docker-compose -f docker/docker-compose.yml up`

Option 2: If the app is already running, exec into the crm-app container and manually apply the migrations:
```
python3 manage.py makemigrations
python3 manage.py migrate
```

2. Test the new endpoint:

- Open your browser and go to the Swagger UI: http://localhost:8000/swagger-ui/#/Dashboard/.
- Log in using your authentication token.
- Enter your organization_id (from the organization table, column id) as a parameter.
- Go to the very bottom of the rerutn of the endpoint. Here you will see the counts of closed deals (won, lost, and total) for each existing month.
